### ISSUE TITLE
Installing cx_Freeze 6.6 from pip fails on mingw-64

### ISSUE THREAD

**Issue by Cimbali** [marcelotduarte/cx_Freeze#1030]
The  compilation fails to find the cx_Logging header(s) included from Win32Service.c

https://github.com/marcelotduarte/cx_Freeze/blob/3ddd59dcc112b15d5145ee2d1bf243f5ada00947/source/bases/Win32Service.c#L13

```
    Running setup.py install for cx-Freeze: started
    Running setup.py install for cx-Freeze: finished with status 'error'
    ERROR: Command errored out with exit status 1:
     command: C:/msys64/mingw64/bin/python3.exe -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'C:/Users/appveyor/AppData/Local/Temp/1/pip-install-arm9wkqz/cx-freeze_1cf019e21f1d4709ab96e57e1598a4bd/setup.py'"'"'; __file__='"'"'C:/Users/appveyor/AppData/Local/Temp/1/pip-install-arm9wkqz/cx-freeze_1cf019e21f1d4709ab96e57e1598a4bd/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' install --record C:/Users/appveyor/AppData/Local/Temp/1/pip-record-f904sh0f/install-record.txt --single-version-externally-managed --compile --install-headers C:/msys64/mingw64/include/python3.8/cx-Freeze
         cwd: C:/Users/appveyor/AppData/Local/Temp/1/pip-install-arm9wkqz/cx-freeze_1cf019e21f1d4709ab96e57e1598a4bd/
    Complete output (49 lines):
    running install
    running build
    running build_py
    creating build
    creating build/lib.mingw-3.8
    creating build/lib.mingw-3.8/cx_Freeze
    copying cx_Freeze/cli.py -> build/lib.mingw-3.8/cx_Freeze
    copying cx_Freeze/common.py -> build/lib.mingw-3.8/cx_Freeze
    copying cx_Freeze/darwintools.py -> build/lib.mingw-3.8/cx_Freeze
    copying cx_Freeze/dist.py -> build/lib.mingw-3.8/cx_Freeze
    copying cx_Freeze/exception.py -> build/lib.mingw-3.8/cx_Freeze
    copying cx_Freeze/executable.py -> build/lib.mingw-3.8/cx_Freeze
    copying cx_Freeze/finder.py -> build/lib.mingw-3.8/cx_Freeze
    copying cx_Freeze/freezer.py -> build/lib.mingw-3.8/cx_Freeze
    copying cx_Freeze/hooks.py -> build/lib.mingw-3.8/cx_Freeze
    copying cx_Freeze/macdist.py -> build/lib.mingw-3.8/cx_Freeze
    copying cx_Freeze/module.py -> build/lib.mingw-3.8/cx_Freeze
    copying cx_Freeze/patchelf.py -> build/lib.mingw-3.8/cx_Freeze
    copying cx_Freeze/setupwriter.py -> build/lib.mingw-3.8/cx_Freeze
    copying cx_Freeze/windist.py -> build/lib.mingw-3.8/cx_Freeze
    copying cx_Freeze/winmsvcr.py -> build/lib.mingw-3.8/cx_Freeze
    copying cx_Freeze/winversioninfo.py -> build/lib.mingw-3.8/cx_Freeze
    copying cx_Freeze/__init__.py -> build/lib.mingw-3.8/cx_Freeze
    copying cx_Freeze/__main__.py -> build/lib.mingw-3.8/cx_Freeze
    creating build/lib.mingw-3.8/cx_Freeze/initscripts
    copying cx_Freeze/initscripts/Console.py -> build/lib.mingw-3.8/cx_Freeze/initscripts
    copying cx_Freeze/initscripts/ConsoleSetLibPath.py -> build/lib.mingw-3.8/cx_Freeze/initscripts
    copying cx_Freeze/initscripts/SharedLib.py -> build/lib.mingw-3.8/cx_Freeze/initscripts
    copying cx_Freeze/initscripts/SharedLibSource.py -> build/lib.mingw-3.8/cx_Freeze/initscripts
    copying cx_Freeze/initscripts/__startup__.py -> build/lib.mingw-3.8/cx_Freeze/initscripts
    running build_ext
    dllwrap: WARNING: dllwrap is deprecated, use gcc -shared or ld -shared instead
    
    creating build/temp.mingw-3.8
    creating build/temp.mingw-3.8/source
    creating build/temp.mingw-3.8/source/bases
    C:\msys64\mingw64\bin/x86_64-w64-mingw32-gcc.exe -Wno-unused-result -Wsign-compare -DNDEBUG -march=x86-64 -mtune=generic -O2 -pipe -fwrapv -D__USE_MINGW_ANSI_STDIO=1 -D_WIN32_WINNT=0x0601 -DNDEBUG -march=x86-64 -mtune=generic -O2 -pipe -fwrapv -D__USE_MINGW_ANSI_STDIO=1 -D_WIN32_WINNT=0x0601 -DNDEBUG -IC:/msys64/mingw64/include/python3.8 -c source/bases/Console.c -o build/temp.mingw-3.8/source/bases/Console.o
    C:\msys64\mingw64\bin/windres.exe -i source/bases/manifest.rc -o build/temp.mingw-3.8/source/bases/manifest.rc.o
    creating build/lib.mingw-3.8/cx_Freeze/bases
    C:\msys64\mingw64\bin/x86_64-w64-mingw32-gcc.exe -s build/temp.mingw-3.8/source/bases/Console.o build/temp.mingw-3.8/source/bases/manifest.rc.o -LC:/msys64/mingw64/lib/python3.8/config-3.8 -LC:/msys64/mingw64/lib -limagehlp -lShlwapi -lpython3.8 -lm -lversion -lshlwapi -o build/lib.mingw-3.8/cx_Freeze/bases/Console.exe -mconsole -municode
    C:\msys64\mingw64\bin/x86_64-w64-mingw32-gcc.exe -Wno-unused-result -Wsign-compare -DNDEBUG -march=x86-64 -mtune=generic -O2 -pipe -fwrapv -D__USE_MINGW_ANSI_STDIO=1 -D_WIN32_WINNT=0x0601 -DNDEBUG -march=x86-64 -mtune=generic -O2 -pipe -fwrapv -D__USE_MINGW_ANSI_STDIO=1 -D_WIN32_WINNT=0x0601 -DNDEBUG -IC:/msys64/mingw64/include/python3.8 -c source/bases/Win32GUI.c -o build/temp.mingw-3.8/source/bases/Win32GUI.o
    C:\msys64\mingw64\bin/windres.exe -i source/bases/manifest.rc -o build/temp.mingw-3.8/source/bases/manifest.rc.o
    C:\msys64\mingw64\bin/x86_64-w64-mingw32-gcc.exe -s build/temp.mingw-3.8/source/bases/Win32GUI.o build/temp.mingw-3.8/source/bases/manifest.rc.o -LC:/msys64/mingw64/lib/python3.8/config-3.8 -LC:/msys64/mingw64/lib -limagehlp -lShlwapi -luser32 -lpython3.8 -lm -lversion -lshlwapi -o build/lib.mingw-3.8/cx_Freeze/bases/Win32GUI.exe -mwindows -municode
    C:\msys64\mingw64\bin/x86_64-w64-mingw32-gcc.exe -Wno-unused-result -Wsign-compare -DNDEBUG -march=x86-64 -mtune=generic -O2 -pipe -fwrapv -D__USE_MINGW_ANSI_STDIO=1 -D_WIN32_WINNT=0x0601 -DNDEBUG -march=x86-64 -mtune=generic -O2 -pipe -fwrapv -D__USE_MINGW_ANSI_STDIO=1 -D_WIN32_WINNT=0x0601 -DNDEBUG -IC:/Users/appveyor/AppData/Local/Temp/1/pip-install-arm9wkqz/cx-freeze_1cf019e21f1d4709ab96e57e1598a4bd/source/bases -IC:/msys64/mingw64/include/python3.8 -c source/bases/Win32Service.c -o build/temp.mingw-3.8/source/bases/Win32Service.o
    source/bases/Win32Service.c:13:10: fatal error: cx_Logging.h: No such file or directory
       13 | #include <cx_Logging.h>
          |          ^~~~~~~~~~~~~~
    compilation terminated.
    error: command 'C:\\msys64\\mingw64\\bin/x86_64-w64-mingw32-gcc.exe' failed with exit status 1
    ----------------------------------------
ERROR: Command errored out with exit status 1: C:/msys64/mingw64/bin/python3.exe -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'C:/Users/appveyor/AppData/Local/Temp/1/pip-install-arm9wkqz/cx-freeze_1cf019e21f1d4709ab96e57e1598a4bd/setup.py'"'"'; __file__='"'"'C:/Users/appveyor/AppData/Local/Temp/1/pip-install-arm9wkqz/cx-freeze_1cf019e21f1d4709ab96e57e1598a4bd/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' install --record C:/Users/appveyor/AppData/Local/Temp/1/pip-record-f904sh0f/install-record.txt --single-version-externally-managed --compile --install-headers C:/msys64/mingw64/include/python3.8/cx-Freeze Check the logs for full command output.
```

To reproduce, simply run `pip install cx_Freeze==6.6` from the msys2 / mingw64 command line on python 3.8.9, windows 10 (10-10.0.17763-SP0) x86_64, cx_Freeze 6.6

**Comment by marcelotduarte** [comment]
I patched cx-logging in msys2 to include the cx_Logging.h
If you install cx-logging using pacman you can build/install cx-freeze using pip

**Comment by Cimbali** [comment]
I think installing cx_Freeze through pacman also works, but it’s installed through pip (as a dependency) when cx_Freeze is installed though pip.

Is cx_Logging not installing a header as/where it should, or is cx_Freeze missing an include path in its compilation process ?

**Comment by marcelotduarte** [comment]
cx-freeze also includes cx_Logging.h to circuvent this error, but it works only with official python. With MinGW it not works.
cx-logging 3.1 will fix this, and I backported the patch to cx-logging in MSYS2.
But if you install cx-logging by pacman it installs .h and .a that cx_Freeze needs to compile using pip.
Another option is: Install the dependencies separately and install cx-freeze using --no-deps

**Comment by marcelotduarte** [comment]
In pympress you can install using pacman. It will work.

**Comment by Cimbali** [comment]
That's what I do now @marcelotduarte, I just thought I would report it in case there was a required fix here. I guess this can be closed now or once cx_Logging 3.1 is out.

**Comment by marcelotduarte** [comment]
In MSYS, cx-logging via pacman has the include file (in version 3.0-2) I include the same patch that I created for cx-logging. For windows cx-freeze 6.6 include it to workaround. So I will close this issue.
